yield_plugin/src/yield_plugin.cpp

```diff
@@ -653,31 +652,58 @@ namespace yield_plugin
-        lanelet::BasicPoint2d object_point(x2,y2);
-        double vehicle_downtrack = wm_->routeTrackPos(vehicle_point).downtrack;
-        double object_downtrack = wm_->routeTrackPos(object_point).downtrack;
-
-        if (vehicle_downtrack > object_downtrack + config_.vehicle_length / 2)  // if half a length of the vehicle past, it is considered behind
-        {
-          RCLCPP_INFO_STREAM(nh_->get_logger(), "Detected an object nearby behind the vehicle at timestamp " << std::to_string(p2a_t));
-          return std::nullopt;
-        }
-        else
-        {
-          RCLCPP_WARN_STREAM(nh_->get_logger(), "Collision detected at timestamp " << std::to_string(p2a_t) << ", x: " << x1 << ", y: " << y1 <<
-            ", within actual downtrack distance: " << object_downtrack - vehicle_downtrack <<
-            ", and collision distance: " << distance);
-          return rclcpp::Time(p2a.header.stamp);
-        }
+        CollisionData collision_data;
+        collision_data.predicted_vehicle_collision_point = lanelet::BasicPoint2d(x1,y1);
+        collision_data.predicted_obstacle_collision_point = lanelet::BasicPoint2d(x2,y2);
+        collision_data.collision_time = rclcpp::Time(p2a.header.stamp);
+        return collision_data;
       }
     }
 
     // No collision detected
     return std::nullopt;
   }
 
+  bool YieldPlugin::is_object_behind_the_vehicle(uint32_t object_id, const CollisionData& collision_data)
+  {
+    const double vehicle_downtrack = wm_->routeTrackPos(collision_data.predicted_vehicle_collision_point).downtrack;
+    const double object_downtrack = wm_->routeTrackPos(collision_data.predicted_obstacle_collision_point).downtrack;
+
+    auto previous_clearance_count = consecutive_clearance_count_for_obstacles_[object_id];
+    // if the object's location is half a length of the vehicle past its rear-axle, it is considered behind
+    // half a length of the vehicle to conservatively estimate the rear axle to rear bumper length
+    if (object_downtrack < vehicle_downtrack - config_.vehicle_length / 2)
+    {
+      consecutive_clearance_count_for_obstacles_[object_id] = std::min(consecutive_clearance_count_for_obstacles_[object_id] + 1, config_.consecutive_clearance_count_for_obstacles_threshold);
+      RCLCPP_INFO_STREAM(nh_->get_logger(), "Detected an object nearby might be behind the vehicle at timestamp: " << std::to_string(collision_data.collision_time.seconds()) <<
+        ", and consecutive_clearance_count_for obstacle: " <<  object_id << ", is: " << consecutive_clearance_count_for_obstacles_[object_id]);
+    }
+    // confirmed false positive for a collision
+    if (consecutive_clearance_count_for_obstacles_[object_id] >= config_.consecutive_clearance_count_for_obstacles_threshold)
+    {
+      RCLCPP_INFO_STREAM(nh_->get_logger(), "Confirmed that the object: " << object_id << " is behind the vehicle at timestamp " << std::to_string(collision_data.collision_time.seconds()));
```

Comment:
Same comment about moving this to calling code.

Reply:
moved out the logging